client/components/swipeable/docs/example.jsx

```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react';
+import Swipeable from 'calypso/components/swipeable';
+
+const SwipeableExample = () => {
+	const [ currentPage, setCurrentPage ] = useState( 0 );
```

Comment:
I've noticed an odd behavior of the example:

* Go to `/devdocs/design`
* Search for "Swipeable"
* Click on the result example.
* You'll end up on `/devdocs/design/swipeable`
* However, you can see all 3 slides on the same page:

![](https://cldup.com/bCQrIAtZP4.png)

If I refresh `/devdocs/design/swipeable`, it all looks and works well.

Reply:
I am able to replicate this as well. I think it has to do with the memoize function storing the width of the elements. So when you go to one that has a smaller width than the current one you end up in a situation where the size is not the same anymore. In order to fix we would need a way to force the re-render of the swipeable component. Or have the component listen to some sort of event that would trigger a rerender. Maybe something like https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API ? Or have a prop just for cases like this.

Reply:
The reason is that the container width between the "components list view" and "component single view" is different, so I think this indicates that we're not properly listening to dimensions changes. What if this happens in a real-world scenario? I think we should fix this. Perhaps we could use the [Resize Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Resize_Observer_API)? It is supported in all major browsers that we support AFAIK.